### ISSUE TITLE
dotnet-xunit doesn't seem to be working properly with Microsoft.AspNetCore.TestHost 1.1.2 on .NET Core 1.1.2

### ISSUE THREAD

**Issue by plitwinski** [xunit/xunit#1486]
I've just spent half of a day on investigating failing tests issue and found out that following errors are happening when running tests with latest dotnet-xunit via console:

- Some parts of json serialized message are not passed to an asp.net core controller when using test server (message is being properly serialized but is missing dictionary property once deserialized)

input class looks as follows:
```
public class ExampleRequest
    {
        [JsonProperty("id")]
        public string Id { get; set; }
        [JsonProperty("params")]
        public IDictionary<string, string> Parameters { get; set; }
    }
```
so serialized version looks like this:
` [{"id":"id001", "params":{"Paramter1":"value1","Paramter2":"value2"}}]`
however, deserialized version ends up like this (only for latest dotnet-xunit versions)
` [{"id":"id001", "params": null }]`
      
- ContentRootPath in IHostingEnvironment parameter is not set properly in Startup class (uses dotnet-xunit directory rather than currently open directory)

All tests are passing if run with dotnet test or test explorer in VS2017 update 3.

I'm using following versions of packages:
```
- xunit - 2.3.0-rc3-build3818
- xunit.core  - 2.3.0-rc3-build3818
- xunit.runner.visualstudio  - 2.3.0-rc3-build3818
- Microsoft.NET.Test.Sdk - 15.0.0
- Microsoft.AspNetCore.TestHost - 1.1.2
- Microsoft.AspNetCore.Mvc - 1.1.3
- Newtonsoft.Json - 10.0.3
```

Can reproduce on both Windows 10 (dev machine) and CentOS 7  (CI agent)

dotnet-xunit versions affected:
```
- 2.3.0-rc3-build3818
- 2.3.0-rc2-build3812
- 2.3.0-rc1-build3809
```

last stable: `2.3.0-beta5-build3769`

**Comment by bradwilson** [comment]
Please provide a repro project.

**Comment by bradwilson** [comment]
> ContentRootPath in IHostingEnvironment parameter is not set properly in Startup class (uses dotnet-xunit directory rather than currently open directory)

You cannot assume that the current directory will be the directory of your test assembly. If you want that to happen, you'll need to ensure it's done yourself.

**Comment by plitwinski** [comment]
That's what I thought (and changed that accordingly in my code base) but it used to set the directory to test assembly's one so I mentioned that too.

Sure, will try to put together a sample repo to reproduce.

**Comment by plitwinski** [comment]
Ok, I looked at it again and finally fixed it by changing property 
```
[JsonProperty("params")]
public IDictionary<string, string> Parameters { get; set; }
```
to
```
[JsonProperty("params")]
public IDictionary<string, string> Params { get; set; }
```
and upgrading all xunit packages to stable `2.3.0` version
I know it's really odd but it seems to be working. Maybe something in my solution overrides Newtosoft.Json configs when running via dotnet xunit (just a long shot)?

I've tried to reproduce the problem in a new solution but without success (all is working fine!) even though I've added all packages (like EF core in-memory database, autofac interceptors, akka.net, test server, etc...).

Guess it can be closed now as I don't know what might be causing this behaviour and cannot reproduce it...
Anyway, thanks a lot for taking your time to look at this issue.
